### PR TITLE
Update version support documentation

### PR DESCRIPTION
**Pull request by sedefsavas**
/kind documentation

**What this PR does / why we need it**:
Update EOL dates and release versioning docs.

**Which issue(s) this PR fixes** *(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close the issue(s) when PR gets merged)*:
Fixes #

**Special notes for your reviewer**:

**Checklist**:

- [ ] squashed commits
- [ ] includes documentation
- [ ] adds unit tests
- [ ] adds or updates e2e tests

**Release note**:

```release-note
NONE
```

### CODE REVIEW

**Comment by sedefsavas** [comment]
@richardcase could you check if I missed anything related to our versioning policy?

File: docs/book/src/topics/reference/versions.md [hunk 1]
```diff
@@ -1,22 +1,55 @@
...
+
+## End-of-Life Timeline
+
+CAPA team maintains branches for **v1.x (v1beta1)**, **v0.7 (v1alpha4)**, and **v0.6 (v1alpha3)**.
+
+CAPA branches follow their compatible Cluster API branch EOL date.
+
+| API Version   | Branch      | Supported Until |
+| ------------- |-------------|-----------------|
+| **v1alpha4**  | release-0.4 | 2022-04-06      |
```
Comment (richardcase) [inline docs/book/src/topics/reference/versions.md:45]:
```suggestion
| **v1alpha4**  | release-0.7 | 2022-04-06      |
```

File: docs/book/src/topics/reference/versions.md [hunk 2]
```diff
@@ -1,22 +1,55 @@
...
+| **v1alpha3**  | release-0.3 | 2022-02-23      |
```
Comment (richardcase) [inline docs/book/src/topics/reference/versions.md:46]:
```suggestion
| **v1alpha3**  | release-0.6 | 2022-02-23      |
```

File: docs/book/src/topics/reference/versions.md [hunk 3]
```diff
@@ -1,22 +1,55 @@
...
+
+## Compatibility with Kubernetes Versions
+
+ CAPA API versions support all Kubernetes versions that is supported by its compatible Cluster API version:
+
+|     API Version              | v1alpha3 (v0.3) | v1alpha4 (v0.4) | v1beta1 (v1.x) |
```
Comment (richardcase) [inline docs/book/src/topics/reference/versions.md:52]:
```suggestion
|     API Version              | v1alpha3 (v0.6) | v1alpha4 (v0.7) | v1beta1 (v1.x) |
```

Comment (sedefsavas) [reply docs/book/src/topics/reference/versions.md]: This row is CAPI API versions. Made it clearer.

**Pull request merged**